src/mod_carboncopy.erl

```diff
@@ -264,16 +260,18 @@ carbon_copy_children(?NS_CC_2, JID, Packet, Direction) ->
                                  attrs = [{<<"xmlns">>, ?NS_FORWARD}],
                                  children = [complete_packet(JID, Packet, Direction)]} ]} ].
 
-enable(Host, U, R, CC) ->
-    ?DEBUG("enabling for ~p/~p", [U, R]),
+enable(JID, CC) ->
+    ?DEBUG("enabling for ~p", [jid:to_binary(JID)]),
     KV = {?CC_KEY, cc_ver_to_int(CC)},
-    {ok, KV} = ejabberd_sm:store_info(U, Host, R, KV),
-    ok.
+    case ejabberd_sm:store_info(JID, KV) of
+        {ok, KV} -> ok;
+        _ -> {error, error}
```

Comment:
I think it'd be better to do the following:
```suggestion
        {error, _} = Err -> Err
```
This way, we'll see the exact error in logs from the `iq_handler` function in this module. Currently the root cause is shadowed by the `{error, error}` tuple.